### PR TITLE
Detangle test environment confusion

### PR DESCRIPTION
**Pull request by rushtong**
Addresses https://broadinstitute.atlassian.net/browse/GAWB-1225
Requires https://github.com/broadinstitute/firecloud-develop/pull/218 to run locally.

---
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -44,51 +44,28 @@ libraryDependencies ++= Seq(
...
-
-addCommandAlias("flywayClean", "safeClean")
+flywayCleanDisabled := true
 
 releaseSettings
 
 shellPrompt := { state => "%s| %s> ".format(GitCommand.prompt.apply(state), version.value) }
 
-javaOptions in Test ++= Seq("-Dconfig.file=" + Option(System.getenv("TEST_CONFIG")).getOrElse("src/test/resources/reference.conf"))
+javaOptions in Test ++= Seq("-Dconfig.file=src/test/resources/reference.conf")
```
Comment (rushtong) [inline build.sbt]: This prevents unit tests from using anything other than the test config so developers don't accidentally wipe out dev/qa/prod

File: build.sbt [hunk 2]
```diff
@@ -44,51 +44,28 @@ libraryDependencies ++= Seq(
...
-                       | `addCommandAlias("flywayClean", "safeClean")`
-                       |
-                       | in the build.sbt file and rerun `sbt flywayClean`
-                       |
-                       | """.stripMargin)
-
-// Wrapper around flywayClean for safety.
-
-addCommandAlias("flywayClean", "safeClean")
+flywayCleanDisabled := true
```
Comment (rushtong) [inline build.sbt:60]: Being extra safe here - we should never have to drop our databases. If we do, that should never be done in an sbt task and instead should be scripted by devops.

File: build.sbt [hunk 3]
```diff
@@ -44,51 +44,28 @@ libraryDependencies ++= Seq(
 //  "org.broadinstitute" %% "cromwell" % "0.9" excludeAll ExclusionRule(organization = "com.gettyimages"),
   "org.broadinstitute.dsde.vault" %% "vault-common" % "0.1-15-bf74315",
   "org.mongodb" %% "casbah" % "2.8.2",
-  "org.flywaydb" % "flyway-core" % "3.2.1",
+  "org.flywaydb" % "flyway-core" % "4.0.3",
+  "org.flywaydb" % "flyway-sbt" % "4.0.3",
```
Comment (rushtong) [inline build.sbt]: Need the updated library to get access to `flywayCleanDisabled` setting.

File: build.sbt [hunk 4]
```diff
@@ -161,8 +138,9 @@ assemblyMergeStrategy in assembly := {
 
 Revolver.settings.settings
 Revolver.enableDebugging(port = 5051, suspend = false)
+
 // When JAVA_OPTS are specified in the environment, they are usually meant for the application
 // itself rather than sbt, but they are not passed by default to the application, which is a forked
 // process. This passes them through to the "re-start" command, which is probably what a developer
 // would normally expect.
-javaOptions in reStart := sys.env("JAVA_OPTS").split(" ")
+javaOptions in reStart ++= sys.env("JAVA_OPTS").split(" ").toSeq
```
Comment (rushtong) [inline build.sbt:147]: See https://github.com/broadinstitute/firecloud-orchestration/commit/42f09b7bf31b62cd5ec8fd918775d4d34890e5d1

File: project/plugins.sbt [hunk 5]
```diff
@@ -1,6 +1,6 @@
 resolvers += "Flyway" at "https://flywaydb.org/repo"
 
-addSbtPlugin("org.flywaydb" % "flyway-sbt" % "3.2.1")
+addSbtPlugin("org.flywaydb" % "flyway-sbt" % "4.0.3")
```
Comment (davidangb) [inline project/plugins.sbt:3]: do you need this in both `build.sbt` and `plugins.sbt`?

Comment (rushtong) [reply project/plugins.sbt:3]: I had compile problems without both ... but I'll double check.

Comment (rushtong) [reply project/plugins.sbt:3]: Turned out the issues I had were unrelated - removed the extra dependency in build.sbt

File: build.sbt [hunk 6]
```diff
@@ -44,51 +44,30 @@ libraryDependencies ++= Seq(
 //  "org.broadinstitute" %% "cromwell" % "0.9" excludeAll ExclusionRule(organization = "com.gettyimages"),
   "org.broadinstitute.dsde.vault" %% "vault-common" % "0.1-15-bf74315",
   "org.mongodb" %% "casbah" % "2.8.2",
-  "org.flywaydb" % "flyway-core" % "3.2.1",
+  "org.flywaydb" % "flyway-core" % "4.0.3",
```
Comment (davidangb) [inline build.sbt]: you can leave this out of build.sbt too; it'll get pulled in as a dependency from the flyway in plugins.sbt

**Review by davidangb — APPROVED** [review]
One minor comment about a build.sbt tweak inline, otherwise good to go. If you make the build.sbt change (and it works for you) and don't change anything else, I don't need to review again. 👍

File: build.sbt (hunk 6 above)
Comment (rushtong) [reply build.sbt]: Yes, that works!

**Pull request merged**